it/test/controllers/optOut/ConfirmOptOutControllerISpec.scala

```diff
@@ -156,7 +169,36 @@ class ConfirmOptOutControllerISpec extends ComponentSpecBase {
         val result = IncomeTaxViewChangeFrontendManageBusinesses.postConfirmOptOut()
 
         result should have(
-          httpStatus(INTERNAL_SERVER_ERROR),
+          httpStatus(SEE_OTHER),
+        )
+
+      }
+    }
+
+    s"user confirms opt-out for multi-year scenario and update fails" should {
+      "show Opt Out error page" in {
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, propertyOnlyResponse)
+
+        val threeYearStatus = ITSAYearStatus(ITSAStatus.Voluntary, ITSAStatus.Voluntary, ITSAStatus.Voluntary)
+        ITSAStatusDetailsStub.stubGetITSAStatusFutureYearsDetailsWithGivenThreeStatus(dateService.getCurrentTaxYearEnd, threeYearStatus)
+        CalculationListStub.stubGetLegacyCalculationList(testNino, previousYear.endYear.toString)(CalculationListIntegrationTestConstants.successResponseNotCrystallised.toString())
+        ITSAStatusUpdateConnectorStub.stubPUTItsaStatusUpdate(propertyOnlyResponse.asInstanceOf[IncomeSourceDetailsModel].nino,
+          BAD_REQUEST, Json.toJson(OptOutUpdateResponseFailure.defaultFailure()).toString(),
+          Map(ITSAStatusUpdateConnector.CorrelationIdHeader -> "123")
+        )
+
+        val newSessionData = UIJourneySessionData(
+          sessionId = hc.sessionId.get.value,
+          journeyType = "OPTOUT",
+          optOutSessionData = Some(OptOutSessionData(Some("2023-2024")))
+        )
+        sessionService.createSession("OPTOUT")
+        sessionService.setMongoData(newSessionData)
+
+        val result = IncomeTaxViewChangeFrontendManageBusinesses.postConfirmOptOut()
+
+        result should have(
+          httpStatus(SEE_OTHER)
```

Comment:
we use SEE_OTHER for non error pages, can add more content asserts to make sure it's an error page